### DISCUSSION TITLE
Reading a ROOT file with multiple cycles of ttrees

### DISCUSSION THREAD

**Discussion by nakolkar16** [scikit-hep/uproot5#921] [Q&A] [answered]
Dear uproot users,

I have a ROOT file with such format: treename;1 treename;2 treename;3 anothertree;13 anothertree;14
Now I want uproot to only consider the latest ttree which is the ttree with the highest number. I am aware that I can call the ttree just by its name and it will get the latest one but what if I want to extract the treename inside a loop. Something like this:
f = uproot.open("file.root")
for tree in f.keys(): 
    do something only with the LATEST ttree not the others! (I want to accomplish this)

Thank you in advance!

**Comment by ekourlit** [comment]
First and simply, to get the last element of the `f.keys()` list you can just do `last_tree = f.keys()[-1]`.  
However, if you have multiple trees with the same name and just an additional ";1", ";2", etc. this is usually something internal to the ROOT files and the user should just call the tree name "treename" without caring about those additional numbers. For example:
```python
with uproot.open("file.root") as f:
    tree = f['treename']
```
If you want to learn more: https://root-forum.cern.ch/t/aux-1-aux-2-aux3/13345

**Reply by jpivarski** [reply]
I guess it's true that the keys will always be in cycle order, and the return value of `keys` is a list, not an iterable, but you might have a situation in which there's more than one object, irrespective of the cycle number.

The `cycle=False` kwarg was made for this.

**Comment by NJManganelli** [comment, marked as answer]
The keys and items methods have a kwarg cycle number, simply set it to false and you will get the keys without cycle number appended. For items, that means you’ll only grab the latest version for that key.

**Reply by nakolkar16** [reply]
Thank you very much for the answer. I used cycle=False kwarg while calling the keys and it grabbed the latest key
